src/abstractdataframe/join.jl

```diff
@@ -290,6 +298,26 @@ function Base.join(df1::AbstractDataFrame,
 
     joiner = DataFrameJoiner(df1, df2, on)
 
+    # Check merge key validity
+    left_invalid = validate[1] ? any(nonunique(joiner.dfl, joiner.left_on)) : false
```

Comment:
I do not remember if it was discussed but `nonunique(joiner.dfl, joiner.left_on)` and `nonunique(joiner.dfr, joiner.right_on)` could be saved in a variable to avoid making the call again on error.